Sources/Nimble/Adapters/NimbleXCTestHandler.swift

```diff
@@ -76,7 +76,8 @@ public func recordFailure(_ message: String, location: SourceLocation) {
 #if SWIFT_PACKAGE
     XCTFail("\(message)", file: location.file, line: location.line)
 #else
-    if let testCase = CurrentTestCaseTracker.sharedInstance.currentTestCase {
+    if let testCase = CurrentTestCaseTracker.sharedInstance.currentTestCase,
+       #available(OSXApplicationExtension 10.13, *) {
```

Comment:
This breaks compatibility for macOS 10.12 or before, so can't be accepted as is. I'll consider another way.

Reply:
currently it's a runtime crash in macOS 10.12

Reply:
And it's even not possible to run a current macOS Version on 10.12. 

As a mac Developer it would be my choice to just simply drop 10.12 support, because you are not running your unit test on macOS 12, because it's not even possible with the current version Xcode.

Reply:
Would it be OK to only break macOS 10.12 compatibility for Swift 5.3 and greater?